File: 백준/Silver/19939. 박 터뜨리기/박 터뜨리기.py
import sys
from itertools import combinations
import math
input = sys.stdin.readline

n,m = map(int, input().split())
## print(n,m)

minBall = (1+m)*m/2
if (n<minBall):
    print(-1)
else:
    n-=minBall
    if (n%m==0):
        print(m-1)
    else:
        print(m)

File: 백준/Silver/19939. 박 터뜨리기/박 터뜨리기.js
const readline = require("readline")
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout

})

const lines = []

rl.on("line", (line) => {

    lines.push(line.trim())
}).on("close", () => {
    let [N, K] = lines[0].split(" ").map(Number)
    const minBall = 0.5 * (1 + K) * K
    let answer
    if (N < minBall) {
        answer = -1
    } else {
        N -= minBall
        if (N % K === 0) {
            answer = K - 1
        } else {
            answer = K
        }
    }
    console.log(answer)

})